From 28b8f04a5256ca5ec0781b06ee9353c37c650980 Mon Sep 17 00:00:00 2001
From: Bruce Allan <bruce.w.allan@intel.com>
Date: Thu, 7 Jan 2010 16:30:56 +0000
Subject: [PATCH] e1000e: call pci_save_state() after pci_restore_state()
References: bnc#578981
Patch-mainline: v2.6.33-rc5

Due to a change in pci_restore_state()[1] which clears the saved_state
flag, the driver should call pci_save_state() to set the flag once again
to avoid issues with EEH (same fix that recently was submitted for ixgbe).

[1] commmit 4b77b0a2ba27d64f58f16d8d4d48d8319dda36ff

Signed-off-by: Bruce Allan <bruce.w.allan@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Brandon Philips <bphilips@suse.de>

---

 drivers/net/e1000e/netdev.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

Index: linux-2.6.32.8-0.0.0.3.1f53358/drivers/net/e1000e/netdev.c
===================================================================
--- linux-2.6.32.8-0.0.0.3.1f53358.orig/drivers/net/e1000e/netdev.c	2010-02-09 12:22:44.000000000 -0500
+++ linux-2.6.32.8-0.0.0.3.1f53358/drivers/net/e1000e/netdev.c	2010-02-10 12:53:58.000000000 -0500
@@ -4653,6 +4653,7 @@
 
 	pci_set_power_state(pdev, PCI_D0);
 	pci_restore_state(pdev);
+	pci_save_state(pdev);
 	e1000e_disable_l1aspm(pdev);
 
 	err = pci_enable_device_mem(pdev);
@@ -4804,6 +4805,7 @@
 	} else {
 		pci_set_master(pdev);
 		pci_restore_state(pdev);
+		pci_save_state(pdev);
 
 		pci_enable_wake(pdev, PCI_D3hot, 0);
 		pci_enable_wake(pdev, PCI_D3cold, 0);
